splice_machine/src/main/java/com/splicemachine/derby/stream/output/update/UpdatePipelineWriter.java

```diff
@@ -95,10 +96,12 @@ public void open(TriggerHandler triggerHandler, SpliceOperation operation, boole
         // Get the DVDS for the primary keys...
         for(int i=0;i<columnOrdering.length;++i)
             kdvds[i]=LazyDataValueFactory.getLazyNull(formatIds[columnOrdering[i]]);
-        colPositionMap=new int[heapList.size()];
+        colPositionMap=new int[heapList.size()*2];
         // Map Column Positions for encoding
-        for(int i=heapList.anySetBit(), pos=heapList.getNumBitsSet();i!=-1;i=heapList.anySetBit(i),pos++)
-            colPositionMap[i]=pos;
+        for(int i=heapList.anySetBit(), pos=heapList.getNumBitsSet();i!=-1;i=heapList.anySetBit(i),pos++) {
+            colPositionMap[i] = pos;
+            colPositionMap[i+heapList.size()] = pos - heapList.getNumBitsSet();
```

Comment:
Can you add a comment about this part? The map being twice as long as before to store old and new values

Reply:
I added a short comment and a reference to `UpdateUtils.extractWriteFromUpdate`